### PR TITLE
Hook queries panel up to real data

### PR DESCRIPTION
**Pull request by robertbrignull**
This PR implements supplying the queries panel with real data from the workspace. We make use of the `Discovery` class, which is already used by the QL Test panel feature and has a very similar set of requirements to us. A lot of inspiration has come directly from the `QLTestDiscovery` class.

Tested locally using the codeql starter workspace:
Screenshot 2023-05-19 at 16 40 08

It successfully handles updating when there's changes to the filesystem, and the automatic remembering of which items are expanded or collapsed seems pretty good.

There are some rough edges that can be improved but I think this is a good starting point. The main thing I'd like to look at in future PRs is single vs multi-root workspaces. In a single-root workspace maybe we hide the top-level item which is just the workspace name.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by shati-patel — APPROVED** [review]
Nice! Approving, since it works locally and it's still hidden behind a feature-flag (but also happy if you'd prefer to wait for another review before merging ❤️)

As discussed offline, there are a few specific things we'd like to fix, but we can do these in follow-up PRs 🧹 
1. Hide the extra folders called `.` from the tree view.
2. Remove the spammy `resolve queries` logs from the CodeQL Extension Log (they currently appear whenever any `.ql` file or containing folder changes).
3. Clean up VS Code imports (e.g. decide whether to import more things from `App`.)
   - This should help us write unit tests.

**Comment by robertbrignull** [comment]
Personally I'm happy to merge now since it's fully hidden and we're still developing and iterating.

**Pull request merged**
